### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by TMcreative1**

### CODE REVIEW

File: src/main/kotlin/com/playwright/remote/engine/download/impl/Download.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.io.InputStream
+import java.nio.file.Path
+
+class Download(parent: ChannelOwner, type: String, guid: String, initializer: JsonObject) : ChannelOwner(
+    parent,
+    type,
+    guid,
+    initializer
+), IDownload {
+    private var browser: IBrowser = (parent as IBrowserContext).browser()
```
Comment (Roman-Mitusov) [inline src/main/kotlin/com/playwright/remote/engine/download/impl/Download.kt:19]: This property of Download class is not used inside the class.
Remove it or make use of it.

**Review by Roman-Mitusov — CHANGES_REQUESTED** [review]

**Comment by Roman-Mitusov** [comment]
Please make sure that our remote API is compatible with current Playwright API. There was a slight changes at least in Download and Video API of Playwright.

**Review by Roman-Mitusov — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/playwright/remote/engine/options/CheckOptions.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+     * element.
+     */
+    var position: Position? = null,
+    /**
+     * Maximum time in milliseconds, defaults to 30 seconds, pass {@code 0} to disable timeout. The default value can be changed by
+     * using the {@link BrowserContext#setDefaultTimeout BrowserContext.setDefaultTimeout()} or {@link Page#setDefaultTimeout
+     * Page.setDefaultTimeout()} methods.
+     */
+    var timeout: Double? = null,
+    private val builder: IBuilder<CheckOptions>
```
Comment (TMcreative1) [inline src/main/kotlin/com/playwright/remote/engine/options/CheckOptions.kt:28]:
```suggestion
    @Transient private val builder: IBuilder<CheckOptions>
```

File: src/main/kotlin/com/playwright/remote/engine/options/element/ScrollIntoViewIfNeededOptions.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import com.playwright.remote.engine.options.api.IBuilder
+
+data class ScrollIntoViewIfNeededOptions @JvmOverloads constructor(
+    /**
+     * Maximum time in milliseconds, defaults to 30 seconds, pass {@code 0} to disable timeout. The default value can be changed by
+     * using the {@link BrowserContext#setDefaultTimeout BrowserContext.setDefaultTimeout()} or {@link Page#setDefaultTimeout
+     * Page.setDefaultTimeout()} methods.
+     */
+    var timeout: Double? = null,
+    private val builder: IBuilder<ScrollIntoViewIfNeededOptions>
```
Comment (Roman-Mitusov) [inline src/main/kotlin/com/playwright/remote/engine/options/element/ScrollIntoViewIfNeededOptions.kt:12]: Please add @Trancient annotation to this field

**Review by Roman-Mitusov — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/playwright/remote/engine/options/element/ScrollIntoViewIfNeededOptions.kt (hunk 3 above)
Comment (TMcreative1) [inline src/main/kotlin/com/playwright/remote/engine/options/element/ScrollIntoViewIfNeededOptions.kt:12]:
```suggestion
    @Transient private val builder: IBuilder<ScrollIntoViewIfNeededOptions>
```

**Review by Roman-Mitusov — APPROVED** [review]

**Pull request merged**